utils/gencorp.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python
+"""
+
+    Greynir: Natural language processing for Icelandic
+
+    Text and parse tree pair generator
+
+    Copyright (C) 2020 Miðeind ehf.
+
+       This program is free software: you can redistribute it and/or modify
+       it under the terms of the GNU General Public License as published by
+       the Free Software Foundation, either version 3 of the License, or
+       (at your option) any later version.
+       This program is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see http://www.gnu.org/licenses/.
+
+
+    This utility program generates sentence trees for GreynirCorpus.
+
+        https://github.com/mideind/GreynirCorpus
+
+    Depends on Miðeind's fork of the Annotald parse tree annotation tool.
+
+        https://github.com/mideind/Annotald
+
+    The output format is similar to that of the Penn Treebank.
+
+"""
+
+import os
+import sys
+import random
+import gc
+
+# Hack to make this Python program executable from the utils subdirectory
+basepath, _ = os.path.split(os.path.realpath(__file__))
+_UTILS = os.sep + "utils"
+if basepath.endswith(_UTILS):
+    basepath = basepath[0 : -len(_UTILS)]
+    sys.path.append(basepath)
+
+from settings import Settings
+from article import Article
+from tree import Tree
+
+# To make this work, clone Miðeind Annotald repo, enter the Greynir
+# virtualenv and run "python setup.py develop" from the Annotald repo root
+from annotald.reynir_utils import reynir_sentence_to_annotree, simpleTree2NLTK
+from annotald.annotree import AnnoTree
+
+
+# Min num tokens in sentence
+MIN_SENT_LENGTH = 3
+
+# Num sentences per batch to shuffle
+MAX_BATCH = 10000
+
+# Separator for sentence trees in output file
+SEPARATOR = "\n\n"
+
+# Skip sentences containing these tokens
+ENGLISH_WORDS = frozenset(
+    [
+        "the",
+        "a",
+        "is",
+        "each",
+        "year",
+        "our",
+        "on",
+        "in",
+        "and",
+        "this",
+        "that",
+        "they",
+        "what",
+        "when",
+        "s",
```

Comment:
Leaving out `s` and `t` may be too much?

Reply:
See previous comment.